### ISSUE TITLE
Bug in `parsebib--convert-tex-italics`

### ISSUE THREAD

**Issue by usaoc** [joostkremers/parsebib#22]
`parsebib--convert-tex-italics` introduced in 83a77ea errors when the submatch has a backslash followed by a character that signals an error in `replace-match`.  An example is the title `"Fexprs as the Basis of {{Lisp}} Function Application, or, {\\emph{\\$vau}}: {{The}} Ultimate Abstraction"`, which causes `replace-match` to signal the error ``"Invalid use of `\\' in replacement text"``.  `parsebib--convert-tex-bold` and `parsebib--convert-tex-small-caps` are also subject to this bug.

**Comment by joostkremers** [comment]
Thanks for your report. While investigating this bug I ran into another bug and interestingly enough, if that second bug hadn't existed, you wouldn't have run into your original bug, at least not with `\$vau`.

That doesn't mean your bug isn't a bug, though. It's just that if the second bug hadn't existed, your bug wouldn't have been triggered by this particular title.

@Hugo-Heagren, pinging you here, because you might find this interesting, too.

The bug that I found is that `(parsebib-clean-TeX-markup "\\$")` doesn't produce the desired output `"$"`. Strangely enough, `(parsebib-clean-TeX-markup "\\$ ")` (note the extra space in the string!) does produce `"$"`. I haven't entirely figured out why, but it must have to do with the fact that `$` in regular expressions is a special character for the end of the string/line.

I (hopefully) fixed this bug in commit 1cd8cb1734770b8ce8b5bc6b2c626f15e2735dbf. This also means that the substring `"\$vau"` won't trigger the original bug any more. That bug still exists, though, as can be seen with something like:

    (parsebib-clean-TeX-markup "\\emph{\\LaTeX}")

Here, the replacement string is `"\\LaTeX`", which contains a backslash followed by a character (`L`) that may not follow a backslash in the replacement of `replace-match`.

I tried several solutions, but they all seem to have their problems. I've now implemented a solution that may not be entirely correct from a technical point of view, but it seems to be the most reliable one. When the replacement is a function, `replace-regexp-in-string` is called with its `LITERAL` argument set, so that backslash is not special. This makes it possible for the replacement text for macros such as `"\emph"` to contain a backslash.

I'm sure this solution will run into its own set of problems, in a whole bunch of corner cases I haven't thought of. But then again, trying to parse LaTeX using regexes is a futile exercise anyway, so... :rofl:
